k8s-tf-cluster/10-clusterconfiguration.tf

```diff
@@ -0,0 +1,139 @@
+/*
+This file defines:
+- GCP Service Account for nodes
+- GKE cluster configuration
+- Node pool configurations
+*/
+
+// Create SA for nodes
+resource "google_service_account" "cluster_node_sa" {
+  project      = data.google_project.project.id
+  account_id   = "sa-${var.cluster_name}"
+  display_name = "Service Account for ${var.cluster_name}"
+}
+
+// Add roles for SA
+resource "google_project_iam_member" "cluster_node_sa_logging" {
+  project = data.google_project.project.id
+  role    = "roles/logging.logWriter"
+  member  = "serviceAccount:${google_service_account.cluster_node_sa.email}"
+}
+resource "google_project_iam_member" "cluster_node_sa_monitoring_viewer" {
+  project = data.google_project.project.id
+  role    = "roles/monitoring.viewer"
+  member  = "serviceAccount:${google_service_account.cluster_node_sa.email}"
+}
+resource "google_project_iam_member" "cluster_node_sa_monitoring_metricwriter" {
+  project = data.google_project.project.id
+  role    = "roles/monitoring.metricWriter"
+  member  = "serviceAccount:${google_service_account.cluster_node_sa.email}"
+}
+
+resource "google_bigquery_dataset" "usage_metering" {
+  dataset_id  = "usage_metering"
```

Comment:
We should include the name of the cluster in the id so we don't have the same dataset for differents clusters.